acs-integration-tests/src/test/java/com/ge/predix/test/utils/NewUaaTestUtil.java

```diff
@@ -0,0 +1,171 @@
+import org.springframework.security.oauth2.common.exceptions.InvalidClientException;
+import org.springframework.security.oauth2.provider.client.BaseClientDetails;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.util.MultiValueMap;
+
+import com.ge.predix.acs.utils.JsonUtils;
+
+public class NewUaaTestUtil {
+
+    private static final JsonUtils JSON_UTILS = new JsonUtils();
+
+    private final OAuth2RestTemplate adminRestTemplate;
+    private final String uaaUrl;
+    private final String tokenUrl;
+    private final ACSRestTemplateFactory acsRestTemplateFactory;
+
+    public NewUaaTestUtil(final OAuth2RestTemplate adminRestTemplate, final String uaaUrl) {
+        this.uaaUrl = uaaUrl;
+        this.tokenUrl = this.uaaUrl + "/oauth/token";
+        this.adminRestTemplate = adminRestTemplate;
+        this.acsRestTemplateFactory = null;
+    }
+
+    public NewUaaTestUtil(final ACSRestTemplateFactory acsRestTemplateFactory, final String uaaUrl) {
+        this.uaaUrl = uaaUrl;
+        this.tokenUrl = this.uaaUrl + "/oauth/token";
+        this.acsRestTemplateFactory = acsRestTemplateFactory;
+        this.adminRestTemplate = this.acsRestTemplateFactory.getOAuth2RestTemplateForUaaAdmin(this.tokenUrl);
+    }
+
+    public void createAcsAdminClient(final List<String> acsZones, final String clientId, final String clientSecret) {
+
+        BaseClientDetails acsAdminClient = new BaseClientDetails();
+        ArrayList<SimpleGrantedAuthority> authorities = new ArrayList<>();
+        authorities.add(new SimpleGrantedAuthority("acs.zones.admin"));
+        authorities.add(new SimpleGrantedAuthority("acs.attributes.read"));
+        authorities.add(new SimpleGrantedAuthority("acs.attributes.write"));
+        authorities.add(new SimpleGrantedAuthority("acs.policies.read"));
+        authorities.add(new SimpleGrantedAuthority("acs.policies.write"));
+        for (String zone : acsZones) {
+            authorities.add(new SimpleGrantedAuthority("predix-acs.zones." + zone + ".admin"));
+            authorities.add(new SimpleGrantedAuthority("predix-acs.zones." + zone + ".user"));
+        }
+
+        acsAdminClient.setAuthorities(authorities);
+        acsAdminClient.setAuthorizedGrantTypes(Collections.singletonList("client_credentials"));
+        acsAdminClient.setClientId(clientId);
+        acsAdminClient.setClientSecret(clientSecret);
+        acsAdminClient.setResourceIds(Collections.singletonList("uaa.none"));
+        createOrUpdateClient(acsAdminClient);
+    }
+
+    public OAuth2RestTemplate createAcsAdminClientAndGetTemplate(final List<String> acsZones) {
+        if (this.acsRestTemplateFactory == null) {
+            throw new IllegalStateException("ACSRestTemplateFactory is null");
+        }
+        String clientId = "admin-" + acsZones.get(0);
+        String clientSecret = "adminSecret-" + acsZones.get(0);
+        createAcsAdminClient(acsZones, clientId, clientSecret);
```

Comment:
Agreed. Split up the method for one client to create zone and one to do zone specific actions.